### PR TITLE
modify signal handling.

### PR DESCRIPTION
**Pull request by yayugu**
Modify signal handling from asynchronous signal handler to synchronous signal handling thread.

To respect async-signal-safe.

### CODE REVIEW

**Comment by ikehara** [comment]
Please remove test/run-tests.travis.sh

File: src/lib/server_app.cc [hunk 1]
```diff
@@ -0,0 +1,167 @@
+/**
```
Comment (m-yoshimo) [inline src/lib/server_app.cc]: Please add license declaration

**Comment by ikehara** [comment]
Thanks.

File: src/lib/server_app.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/**
```
Comment (m-yoshimo) [inline src/lib/server_app.h]: Please add license declaration

File: test/lib/test_server_app.cc [hunk 3]
```diff
@@ -0,0 +1,148 @@
+/**
```
Comment (m-yoshimo) [inline test/lib/test_server_app.cc]: Please add license declaration

**Comment by yayugu** [comment]
Rebase branch.
see:
https://github.com/gree/flare/pull/102

**Pull request closed without merging**